internal/app/backend/backend_service.go

```diff
@@ -201,20 +196,16 @@ func callGrpcMmf(ctx context.Context, cc *rpc.ClientCache, profile *pb.MatchProf
 	var conn *grpc.ClientConn
 	conn, err := cc.GetGRPC(address)
 	if err != nil {
-		logger.WithFields(logrus.Fields{
-			"error":    err.Error(),
-			"function": address,
-		}).Error("failed to establish grpc client connection to match function")
-		return status.Error(codes.InvalidArgument, "failed to connect to match function")
+		return status.Error(codes.InvalidArgument, "failed to establish grpc client connection to match function")
 	}
 	client := pb.NewMatchFunctionClient(conn)
 
 	stream, err := client.Run(ctx, &pb.RunRequest{Profile: profile})
 	if err != nil {
-		logger.WithError(err).Error("failed to run match function for profile")
+		err = errors.Wrap(err, "failed to run match function for profile")
 		if ctx.Err() != nil {
 			// gRPC likes to suppress the context's error, so stop that.
-			return ctx.Err()
+			err = errors.Wrap(err, ctx.Err().Error())
```

Comment:
Because if ctx.Err() exists we return it and ignore err after client.Run which is already not nil, that is why I decided to add ctx.Err() to the scope of an error above. We agreed not to log anything on this level that is why I think it's ok to wrap everything and return at once. What do you think?

Reply:
Ok, so the additional info here is that when match functions are canceled, the context is canceled.  It's the error we actually care about, but gRPC suppresses the error and returns something less useful.  We know that if the context is canceled, that is the more important error to show the user.  So it's reasonable to ignore the other error returned, because it's likely to be less useful, and worse, possibly misleading. So please leave this alone (here in and in the other place).

Reply:
Updated, please check again.